src/docs/src/whatsnew/3.4.rst

```diff
@@ -0,0 +1,236 @@
+* :ghissue:`4604`: 6x speedup for common `mem3:dbname/1`` function.
+* :ghissue:`4603`: Update ``vm.args`` settings.
+* :ghissue:`4598`: Speed up internal replicator.
+* :ghissue:`4507`, :ghissue:`4525`: Add more prometheus metrics.
+* :ghissue:`4505`: Trreat JavaScript internal errors as fatal.
+* :ghissue:`4503`: Add ``error_info`` clause for ``410 Gone``.
+* :ghissue:`4494`: Treat single-element keys as key.
+* :ghissue:`4473`: Avoid re-compiling filter view functions.
+* :ghissue:`4401`: Enforce doc ids ``_changes`` filter optimisation limit and
+  raise it from ``100`` to ``1000``.
+* :ghissue:`4394`: Mango: push ``fields`` selection down to data nodes.
+
+Bugfixes
+--------
+
+* :ghissue:`4837`: Fix update bug in ``ets_lru``.
+* :ghissue:`4794`: Fix incorrect raising of ``database_does_not_exist`` error.
+* :ghissue:`4784`: Fix parsing of node name from ``ERL_FLAGS1``
+  in ``remsh``.
+* :ghissue:`4782`, :ghissue:`4891`: Mango: prevent occasional
+  duplication of paginated text results.
+* :ghissue:`4781`: Fix badrecord error when replicator is logging http usage.
+* :ghissue:`4759`: TLS: use HTTP rules for hostname verification.
+* :ghissue:`4758`: Remove sensitive headers from the mochiweb request in pdict.
+* :ghissue:`4751`: Mango: correct behavour of ``fields`` on ``_explain``.
+* :ghissue:`4722`: Fix badmatch error when purge requests time out.
+* :ghissue:`4716`: Fix pending count for reverse changes feed.
+* :ghissue:`4709`: Mango: improve handling of invalid fields.
+* :ghissue:`4703`, :ghissue:`4703`: Fix empty facet search results.
+* :ghissue:`4682: ``_design_doc/queries`` with ``keys`` should only return
+  design docs.
+* :ghissue:`4669`: Allow for more than two replicator socket options.
+* :ghissue:`4666`: Improve error handling in config API.
+* :ghissue:`4659`: Mango: remove duplicates from ``indexable_fields/1``
+  results.
+* :ghissue:`4658`: Fix undefined range in mem3_rep purge replication logic.
+* :ghissue:`4653`: Fix ability to use ``;`` inside of config values.
+* :ghissue:`4629`: Fix prometheus to survive ``mem3_sync`` termination.
+* :ghissue:`4626`: Fix purge infos replicating to the wrong shards
+  during shard splitting.
+* :ghissue:`4602`: Fix error handling for the _index endpoint and
+  document ``_index/_bulk_delete``.
+* :ghissue:`4594`: Mango: covering indexes crash on partitioned
+  databases.
+* :ghissue:`4555`: Fix race condition when creating indexes.
+* :ghissue:`4524`: Querying ``_all_docs`` with non-string key should
+  return an empty list.
+* :ghissue:`4514`: GET invalid path under ``_index`` should not cause
+  500 response.
+* :ghissue:`4509`: Make ``remsh`` work with quoted cookie.
+* :ghissue:`4491`: Fix ``couch_index`` to avoid crashes under certain
+  conditions.
+* :ghissue:`4485`: Catch and log any error from ``mem3:local_shards``.
+* :ghissue:`4473`: Fix prometheus counter metric naming.
+* :ghissue:`4458`: Fix text index selection for queries with ``$regex``.
```

Comment:
Since we have the "Mango" prefix for other Mango-related changes (which I like), we should add it there as well.  (I know that the original title PR might not have it, but here is a the chance to fix that up.)